src/include/execution/compiler/operator/output_translator.h

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <utility>
+#include "execution/ast/context.h"
+#include "execution/compiler/function_builder.h"
+#include "execution/compiler/operator/operator_translator.h"
+#include "execution/exec/output.h"
+
+namespace terrier::execution::compiler {
+/**
+ * Consumer that generates code for outputting to upper layers.
+ */
+class OutputTranslator : public OperatorTranslator {
+ public:
+  /**
+   * Construction
+   * @param codegen The code generator
+   */
+  explicit OutputTranslator(CodeGen *codegen);
+
+  void Produce(FunctionBuilder *builder) override;
+  void Abort(FunctionBuilder *builder) override {}
+  void Consume(FunctionBuilder *builder) override;
+
+  // Does nothing
+  void InitializeStateFields(util::RegionVector<ast::FieldDecl *> *state_fields) override {}
+
+  // Create the output struct
+  void InitializeStructs(util::RegionVector<ast::Decl *> *decls) override;
+
+  // Does nothing
+  void InitializeHelperFunctions(util::RegionVector<ast::Decl *> *decls) override {}
+
+  // Does nothing
+  void InitializeSetup(util::RegionVector<ast::Stmt *> *setup_stmts) override {}
+
+  // Does nothing
+  void InitializeTeardown(util::RegionVector<ast::Stmt *> *teardown_stmts) override {}
+
+  // Should never called since this the last layer.
+  ast::Expr *GetOutput(uint32_t attr_idx) override { UNREACHABLE("Should not be called on this translator"); }
```

Comment:
The query's output will be placed in the `OutputBuffer` of the execution context.
This `GetOutput` is used by parent translators to get their children's outputs. Because this is the final layer, it does not have a parent. So this function should not be called.